runtime/lua/provider/health.lua

```diff
@@ -0,0 +1,55 @@
+local M = {}
+
+-- Returns true if `cmd` exits with success, else false.
+local function cmd_ok(cmd)
+  vim.fn.system(cmd)
+  return vim.v.shell_error == 0
+end
+
+local function executable(exe)
+  return vim.fn.executable(exe) == 1
+end
+
+local function empty(s)
+  return vim.trim(s) == ''
```

Comment:
I'd use `s:find('^%s*$') ~= nil` as is. If inside wrapper function, `is_blank()` is better: "empty" <-> `""`, blank <-> "consists from whitespace".

Reply:
Good call on the "is_blank". What's the benefit of using `:find` compared to vim.trim?

Reply:
`find` is faster than calling `match` and checking the output.

Reply:
That's good enough for me :)